packages/yarnpkg-core/sources/execUtils.ts

```diff
@@ -85,15 +85,15 @@ export async function pipevp(fileName: string, args: Array<string>, {cwd, env =
       reject(error);
     });
 
-    child.on(`close`, (code: number, sig: string) => {
+    child.on(`close`, (code, sig) => {
       if (--sigintRefCount === 0)
         process.off(`SIGINT`, sigintHandler);
 
       if (end === EndStrategy.Always || (end === EndStrategy.ErrorCode && code > 0))
         closeStreams();
 
       if (code === 0 || !strict) {
-        resolve({code});
+        resolve({code: getExitCode(code, sig)});
```

Comment:
I think it should instead enter the rejection path

Reply:
That's possible, but I'd find it a bit weird that nonzero exit codes require `strict: true` to reject the promise while signals don't?

There are signals that are not user based, e.g. SIGABRT if the program aborts (e.g. a failed `assert` in C, or an out of memory error in `malloc`) or SIGSEGV for segmentation faults. I'm not sure if differentiating between these and a regular non-zero exit code is useful.

Reply:
You're right, I misread the diff 😃